pages/api/hold/request/[id]/edd.ts

```diff
@@ -1,23 +1,43 @@
 import type { NextApiRequest, NextApiResponse } from "next"
 
-import { postEDDRequest } from "../../../../../src/server/api/hold"
+import {
+  postEDDRequest,
+  fetchPatronEligibility,
+} from "../../../../../src/server/api/hold"
 import { BASE_URL, PATHS } from "../../../../../src/config/constants"
 
 /**
  * Default API route handler for EDD requests
  */
 async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (req.method !== "POST") {
-    res.status(500).json({
+    return res.status(500).json({
       error: "Please use a POST request for the EDD Request API endpoint",
     })
   }
 
   try {
+    const { patronId, jsEnabled, ...rest } = JSON.parse(req.body)
     const holdId = req.query.id as string
+
     const [, itemId] = holdId.split("-")
 
-    const { jsEnabled, ...rest } = JSON.parse(req.body)
+    const patronEligibilityStatus = await fetchPatronEligibility(patronId)
+
+    if (patronEligibilityStatus.status === 401) {
+      switch (jsEnabled) {
+        case true:
+          return res.status(401).json({
+            patronEligibilityStatus,
+          })
+
+        // Server side redirect on ineligibility when Js is disabled
+        // TODO: Move this to seaprate API route
```

Comment:
Why do you want to move this to a separate API route?

Reply:
To be able to remove the jsEnabled flag from the client call.